core-common/src/main/java/org/zalando/nakadi/repository/kafka/AvroDeserializerWithSequenceDecoder.java

```diff
@@ -28,7 +35,11 @@ public AvroDeserializerWithSequenceDecoder(final AvroSchema schemas) {
             metadata.put("received_at", new DateTime(
                     (long) metadata.get("received_at"), DateTimeZone.UTC).toString());
 
-            final GenericRecord event = eventSequenceDecoder.read(envelop.getPayload());
+            final SequenceDecoder eventDecoder = eventSequenceDecoders.computeIfAbsent(
+                    metadata.get("schema_version").toString(),
+                    (v) -> new SequenceDecoder(schemas.getEventTypeSchema(eventTypeName, v)));
```

Comment:
it looks like the sequence decoder does not need to know schema for specific even type. it looks like it would help it to work with all available event type schemas and decide on the fly which one to use.

Reply:
Well, in the current implementation it does need to know the schema, so that it can initialize the datum reader at construct time.  If we try to make it not depend on the schema, we would just move this complexity from here to that class.

Reply:
I mean, you have event type inside metadata

Reply:
so it should be sufficient to know how to deserialise inside the deserialiser

Reply:
or near it, but the point is that you do not need to pass the event type from outside